antlr_ast/ast.py

```diff
@@ -460,15 +456,39 @@ def transformer_method(self, node):
         return transformer_method
 
 
+class BaseNodeRegistry:
+    def __init__(self):
+        self.dynamic_node_classes = {}
+
+    def get_cls(self, cls_name: str, field_names: tuple) -> Type[BaseNode]:
+        """"""
+        if cls_name not in self.dynamic_node_classes:
+            self.dynamic_node_classes[cls_name] = BaseNode.create_cls(
+                cls_name, field_names
+            )
+        return self.dynamic_node_classes[cls_name]
+
+    def isinstance(self, instance: BaseNode, class_name: str) -> bool:
+        """Check if a BaseNode is an instance of a registered dynamic class"""
+        if isinstance(instance, BaseNode):
+            klass = self.dynamic_node_classes.get(class_name, None)
```

Comment:
I'm guessing `klass` is used to avoid conflicts with built-in `class`. Usually, to avoid conflicts, you can replace `class` by `_class`.

Reply:
I used `klass` as pythonwhat also used that.
But I think `class_` would be better (ending with the underscore because it doesn't remind of it being something private).

Reply:
I've seen both cases used, but I agree `class_` is less confusing.